pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2006,13 +2007,29 @@ func buildTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType
 		},
 	}
 
+	if config.Tracing != nil && proxyType == model.SidecarProxy {
+		buildSidecarTracingConfig(config, proxyType, tracingCfg)
```

Comment:
can you comment why this is for sidecars only?

Reply:
My rationale is that gateways serve incoming client requests. It's possible that client input can present issues (e.g. filling up tracing backend causing fewer traces to be stored).

@douglas-reid and @nrjpoddar thoughts? I'll add some comments when we agree to a final decision.

Reply:
Custom tags should not be exposed to clients as there's a possibility of abuse/ddos.

Max path length otoh should be applied to all proxies (sidecars + gateways).